### ISSUE TITLE
The Luminosity ratio of Table data text to its background color is less than required ratio i.e; 4.267:1: A11y_WinForm.NETRuntime_Contrast(Minimum).

### ISSUE THREAD

**Issue by shindepra** [dotnet/winforms#13759]
"Please do not close this bug. This bug should only be closed by TCS, C+AI Accessibility or the Divisional Driver."

### GitHub Tags:
#A11ySev2; #A11yTCS; #WCAG1.4.3; #Win32; #DesktopApp; #Win11; #WinForms .NET Runtime; #BM_WinForms.NETRuntime_Win32_June2025; #Contrast(Minimum); #A11yMAS

### Environment Details:
Microsoft Windows 11 Enterprise (24H2) Build on 26100.4061  

### Repro Steps:

1. Open SQL Server Migration Assistant for oracle.  
2. Splash screen appears and SSMA is launched  
3. Navigate to connect to oracle and connect to target button and connect it.  
4. Navigate to 'Tools' menu and select it.  
5. Navigate to 'Convert schema' menu item and select it.  
6. Navigate all controls after selecting 'Convert schema'.  
7. Verify whether luminosity ratio is sufficient or not for table data text with its background colour.  

### Actual Result:
The Luminosity ratio of Table data text to its background color is less than required ratio ie; 4.267:1  

### Expected Result:
The Luminosity ratio of table data text to its background color should be at least 4.5:1 or more .

### User Empact:
A luminosity ratio of 4.267:1 for table data text falls below the required minimum contrast ratio of 4.5:1 for normal text, making it difficult for users with low vision or color blindness to read the content. This can hinder information access and fails to meet WCAG 2.1 AA contrast requirements.

Sample App used:-

[[[[RadioButtonFocusDemo.zip](https://github.com/user-attachments/files/21483118/RadioButtonFocusDemo.zip)](https://microsoftapc-my.sharepoint.com/:u:/g/personal/v-kkumar2_microsoft_com/ETnjjCF0jsZFiF_P9fDDmyIBlb80rXpmeIDu2or6768kVg?e=M4uY2O)](https://microsoftapc-my.sharepoint.com/:u:/g/personal/v-kkumar2_microsoft_com/ETnjjCF0jsZFiF_P9fDDmyIBlb80rXpmeIDu2or6768kVg?e=cCZLff)](https://microsoftapc-my.sharepoint.com/:u:/g/personal/v-kkumar2_microsoft_com/ETnjjCF0jsZFiF_P9fDDmyIBlb80rXpmeIDu2or6768kVg?e=wBZc6R)

### Attachment:

![Image](https://github.com/user-attachments/assets/2f9e5580-d31a-48ab-a129-215c0f7eaf3d)

**Comment by LeafShi1** [comment]
@merriemcgaw The contrast in **.NET Framework** projects is indeed insufficient.

<img width="1426" height="998" alt="Image" src="https://github.com/user-attachments/assets/6a3b3c45-dc4c-4336-938e-c2f98520f6fe" />

In **.NET** projects, when the mouse is over the brightest part of the text, the contrast reaches 4.498:1, almost reaching 4.5:1. Does this issue need further improvement?

<img width="1476" height="1048" alt="Image" src="https://github.com/user-attachments/assets/5b1817cb-41ec-4f5a-b3fa-91ea451870a5" />

The previous [DataGridViewLinkCell contrast issue](https://github.com/dotnet/winforms/issues/3883) was fixed and only reached 4.498:1

**Comment by merriemcgaw** [comment]
We need to see if we have an open issue in our internal AzDO instance for .NET Framework. If .NET 10 is compatible we should close this issue and track it internally for .NET Framework. This is not an issue we can commit to fixing  in a particular timeframe in .NET Framework unfortunately.

For .NET 10, I consider this to be rounding-error level of a problem, so I think we can consider this as passing. @shindepra please confirm that  4.498:1 is close enough to 4.5:1 that you can close this issue.

**Comment by LeafShi1** [comment]
There are no open issues in our .NET Framework internal AzDO instance, so an internal bug https://devdiv.visualstudio.com/DevDiv/_workitems/edit/2542754 has been filed to track this issue

**Comment by merriemcgaw** [comment]
@LeafShi1  - Can the developer just specify a font color when selected in order to meet the contrast ratio in the .NET Framework application?

**Comment by LeafShi1** [comment]
> Can the developer just specify a font color when selected in order to meet the contrast ratio in the .NET Framework application?

Yes, developers can use `DefaultCellStyle` or the style properties of `Rows/Columns/Cells` to set the color of the selected state

```
dataGridView1.DefaultCellStyle.SelectionForeColor = Color.Yellow;
dataGridView1.DefaultCellStyle.SelectionBackColor = Color.Black;
```

```
dataGridView1.Columns[0].DefaultCellStyle.SelectionForeColor = Color.White;
dataGridView1.Columns[0].DefaultCellStyle.SelectionBackColor = Color.DarkBlue;
```

```
 dataGridView1.Rows[0].Cells[0].Style = new System.Windows.Forms.DataGridViewCellStyle
 {
     SelectionForeColor = Color.White,
     SelectionBackColor = Color.DarkGreen
 };
```

**Comment by merriemcgaw** [comment]
@LeafShi1 can you confirm the ratio in .NET 8? If it's what it is in 10 we won't change anything but if it's less we can consider servicing to support SSMA.

**Comment by LeafShi1** [comment]
The ratio is enough in .NET 8

<img width="2035" height="1272" alt="Image" src="https://github.com/user-attachments/assets/29d3ed78-0288-4252-801d-0f68c9975bf9" />
